ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -443,20 +449,22 @@ def get_sarif_content(report, clang_tidy_version):
         for error in report[filename]:
             description, rule_id = filter(
                 None, re.split('\[|\]', error['error_msg']))
-            rules.append({
+            new_rule = {
                 'id': rule_id,
-                'shortDescription': {
-                    'text': description.rstrip()
-                },
                 'helpUri': 'https://clang.llvm.org/extra/clang-tidy/checks/list.html',
-            })
+            }
+            if new_rule not in rules:
+                rules.append(new_rule)
```

Comment:
How much overlap does this operation have with the deduplication post-processing in process_sarif?

I would prefer to keep the ament output as "raw" as possible from the wrapped tool.

Reply:
The duplication that `process_sarif` handles is within the `results` field. This is preventing duplication on the rules types reported by `clang_tidy`, which ends up in the `runs[0].tool.rules` field of the SARIF. I could see an argument for porting this over to `process_sarif`, but I don't believe this is duplicating work.

Reply:
I don't think the rules should be repeated in the SARIF file. Multiple discovered issues can reference the same rule.